src/app/pages/courses/courses.page.html

```diff
@@ -1,15 +1,27 @@
-<app-search-input [formControl]="searchControl"></app-search-input>
+<app-search-input
+  [formControl]="searchControl"
+  (search)="onSearchClick()"
+></app-search-input>
 <div class="add-course-container">
   <button (click)="onAddItem()">
     <fa-icon [icon]="addCourseIcon"></fa-icon>Add course
   </button>
 </div>
-<ng-container *ngFor="let course of courses$ | async">
-  <app-course-card
-    [course]="course"
-    (delete)="onCardDelete($event)"
-  ></app-course-card>
+<ng-container *ngIf="courses$ | async as courses; else noCourses">
```

Comment:
you could add filter in observable to filter out empty array and remove one ng-container, will simplify things